### PR TITLE
[EFR32] Add Board Support for efr32 boards

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Problem
Some boards were not supported for Matter

#### Change overview
Add board support  for BRD2703A and BRD4162A

**Platform Specific changes** 
Add two more boards support since current Matter SDK is too restrictive for our available chip.
These Boards are more widely available and will enable a greater Matter development for 1.0

#### Testing
Manual tests with BRD2703A and BRD4162A boards

### CODE REVIEW

File: third_party/silabs/efr32_board.gni [hunk 1]
```diff
@@ -75,8 +75,11 @@ if (efr32_board == "BRD4304A") {
 } else if (efr32_board == "BRD2601B") {
   efr32_family = "efr32mg24"
   efr32_mcu = "EFR32MG24B310F1536IM48"
+} else if (efr32_board == "BRD2703A") {
```
Comment (jmartinez-silabs) [inline third_party/silabs/efr32_board.gni:78]: didn't you have to add BRD4162A in here too

**Comment by andy31415** [comment]
@mkardous-silabs - could you please add to the description the reason why this is a required/acceptable 1.0 change?

Master changes is supposed to now accept:
  - spec compliance issues
  - critical security fixes
  - platform fixes (if a platform fix impacts another platform or the core sdk, it shall be more carefully scrutinized)
  - core SDK fixes that have significant value (memory leak, crash, latency/performance, interop, ...)
  - fixes meant to improve testing and/or testability
  - common cluster logic that have significant value
  
 This PR seems to be somewhere in 'platform specific' (but it is not a platform fix directly ...) and interop (new boards added ... so chip working on additional boards ... not sure if that is interop)

File: third_party/silabs/efr32_board.gni (hunk 1 above)
Comment (mkardous-silabs) [reply third_party/silabs/efr32_board.gni:81]: done

**Comment by mkardous-silabs** [comment]
> @mkardous-silabs - could you please add to the description the reason why this is a required/acceptable 1.0 change?
> 
> Master changes is supposed to now accept:
> 
> * spec compliance issues
> * critical security fixes
> * platform fixes (if a platform fix impacts another platform or the core sdk, it shall be more carefully scrutinized)
> * core SDK fixes that have significant value (memory leak, crash, latency/performance, interop, ...)
> * fixes meant to improve testing and/or testability
> * common cluster logic that have significant value
> 
> This PR seems to be somewhere in 'platform specific' (but it is not a platform fix directly ...) and interop (new boards added ... so chip working on additional boards ... not sure if that is interop)

@andy31415 
The PR is 100% platform specific to efr32. The only thing it does is add support for 2 extra boards. We would like this PR to be merged to enable more teams to be able to test matter since not all sites have access to some boards.

**Comment by jepenven-silabs** [comment]
@andy31415 Updated the Comments as requested.

**Review by jmartinez-silabs — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**